application-mocca-calendar-ui/src/main/resources/MoccaCalendar/Code/Macro.xml

```diff
@@ -896,6 +899,71 @@ require(['jquery'], function($) {
     var calendarPopup = relatedTarget.data('calendarPopup');
     new XWiki.MoccaCalendar.MoccaCalendarPopup(calendarPopup.interactionParameters, calendarPopup.helper);
   });
+
+  // Trigger the upload manually when the button is clicked
+  $(document).on('click', '#importCalendarFileButton', function(event) {
+    var form = $('#importForm').get(0);
+    if (event) {
+      event.preventDefault();
+    }
+    startUploading(form);
+  });
+
+  // Start uploading this file, creating a new XHR object with the file data.
+  var startUploading = function (form) {
+    var formData = new FormData();
```

Comment:
Why not using https://api.jquery.com/serialize/ and the `$.post`?

Reply:
Used this method as it was in HTML5: https://github.com/xwikisas/xwiki-platform/blob/03d3ebeaea072fdc4e17f5aff0a5f7c0e5e6de8f/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/widgets/upload.js#L187